From 5c9ea8ef8d9b2aa88a2b52d7e139a1f84f9a1ca2 Mon Sep 17 00:00:00 2001
From: Alexander Sverdlin <alexander.sverdlin@gmail.com>
Date: Sun, 21 Feb 2021 12:06:32 +0100
Subject: [PATCH 4/5] linuxspi: Support inverted GPIO pin

Clear the inversion mask on request and set default state to avoid short
glitches on the GPIO line.
---
 avrdude/linuxspi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/avrdude/linuxspi.c b/avrdude/linuxspi.c
index 6675f48..da2821f 100644
--- a/avrdude/linuxspi.c
+++ b/avrdude/linuxspi.c
@@ -164,7 +164,8 @@ static int linuxspi_open(PROGRAMMER *pgm, char *port)
 
     strcpy(req.consumer_label, progname);
     req.lines = 1;
-    req.lineoffsets[0] = pgm->pinno[PIN_AVR_RESET];
+    req.lineoffsets[0] = pgm->pinno[PIN_AVR_RESET] & ~PIN_INVERSE;
+    req.default_values[0] = !!(pgm->pinno[PIN_AVR_RESET] & PIN_INVERSE);
     req.flags = GPIOHANDLE_REQUEST_OUTPUT;
 
     ret = ioctl(fd_gpiochip, GPIO_GET_LINEHANDLE_IOCTL, &req);
-- 
2.29.2

